### PR TITLE
Fix forced to Double and optimization error hint for top range

### PR DESCRIPTION
**Pull request by zyxxoo**
fix error hit and Integer case to Double

### CODE REVIEW

File: hugegraph-core/src/main/java/com/baidu/hugegraph/traversal/algorithm/SingleSourceShortestPathTraverser.java [hunk 1]
```diff
@@ -212,11 +212,19 @@ private double edgeWeight(HugeEdge edge) {
                 !edge.property(this.weight).isPresent()) {
                 edgeWeight = 1.0;
             } else {
-                edgeWeight = edge.value(this.weight);
+                edgeWeight = toDouble(edge.value(this.weight));
```
Comment (Linary) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/traversal/algorithm/SingleSourceShortestPathTraverser.java:215]: add this.

**Review by Linary — CHANGES_REQUESTED** [review]

**Review by zhoney — APPROVED** [review]

**Review by Linary — APPROVED** [review]

File: hugegraph-core/src/main/java/com/baidu/hugegraph/traversal/algorithm/NeighborRankTraverser.java [hunk 2]
```diff
@@ -244,7 +244,8 @@ private Ranks contributeNewLayer(List<Adjacencies> adjacencies,
         public Step(HugeGraph g, Directions direction, List<String> labels,
                     long degree, long skipDegree, int top, int capacity) {
             E.checkArgument(top > 0 && top <= MAX_TOP,
-                            "The top of each layer can't exceed %s", MAX_TOP);
+                            "The top of each layer must be in (0, %s]",
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/traversal/algorithm/NeighborRankTraverser.java:247]: perfect if prompt the actual top value

File: hugegraph-core/src/main/java/com/baidu/hugegraph/traversal/algorithm/SingleSourceShortestPathTraverser.java [hunk 3]
```diff
@@ -212,11 +212,19 @@ private double edgeWeight(HugeEdge edge) {
                 !edge.property(this.weight).isPresent()) {
                 edgeWeight = 1.0;
             } else {
-                edgeWeight = edge.value(this.weight);
+                edgeWeight = this.toDouble(edge.value(this.weight));
             }
             return edgeWeight;
         }
 
+        private double toDouble(Object obj) {
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/traversal/algorithm/SingleSourceShortestPathTraverser.java:220]: set static

File: hugegraph-core/src/main/java/com/baidu/hugegraph/traversal/algorithm/SingleSourceShortestPathTraverser.java [hunk 4]
```diff
@@ -212,11 +212,19 @@ private double edgeWeight(HugeEdge edge) {
...
+            if (obj instanceof Number) {
+                return ((Number) obj).doubleValue();
+            }
+
+            return (double) obj;
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/traversal/algorithm/SingleSourceShortestPathTraverser.java:225]: prefer to call NumericUtil.toNumber(obj).doubleValue()

**Pull request merged**